File: Spang-PC_C-sharp/Spang-PC_C-sharp/IConnection.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Net;

namespace Spang_PC_C_sharp
{
    interface IConnection
    {
        /// <summary>
        /// Send a message using the UDP-protocoll.
        /// </summary>
        /// <param name="data">The Message</param>
        void SendUDP(byte[] data);

        /// <summary>
        /// Send a message using the TCP-protocoll.
        /// </summary>
        /// <param name="data">The message</param>
        void SendTCP(byte[] data);

        /// <summary>
        /// Recives a UDP message.
        /// </summary>
        /// <returns>The message recived.</returns>
        byte[] ReciveUDP();

        /// <summary>
        /// Recives a TCP message.
        /// </summary>
        /// <returns>The message recived.</returns>
        byte[] ReciveTCP();

        /// <summary>
        /// Gets or sets the time TCP send/recive will block.
        /// </summary>
        int Timeout { get; set; }

        /// <summary>
        /// Closes the connection.
        /// </summary>
        void Close();

        /// <summary>
        /// Gets the connected status.
        /// </summary>
        bool Connected { get; }

        /// <summary>
        /// The remote endpoint the connection is connected to.
        /// </summary>
        IPEndPoint RemoteEndPoint { get; }

        /// <summary>
        /// The local endpoint the connection is connected from.
        /// </summary>
        IPEndPoint LocalEndPoint { get; }
    }
}

File: Spang-mobile/src/network/IConnection.java
package network;

import java.net.InetSocketAddress;

/**
 * Interface to handle the connection at the most basic level.
 * 
 * @author Lukas Kurtyan & Joakim Johansson
 *
 */
public interface IConnection {
		
	/**
	 * Sends the supplied data using the UDP-protocol. 
	 * @param data the data to be sent.
	 */
	void sendUDP(byte[] data);
	
	/**
	 * Sends the supplied data using the TCP-protocol.
	 * @param data
	 */
	void sendTCP(byte[] data);
	
	/**
	 * Receives an array of byte data from the UDP-protocol.
	 * @return the array of byte that was sent over the network.
	 */
	byte[] reciveUDP();
	
	
	/**
	 * Receives an array of byte data from the TCP-protocol.
	 * @return the array of byte that was sent over the network.
	 */
	byte[] reciveTCP();

	/**
	 * Is the connected connected.
	 * @return connection status.
	 */
	boolean isConnected();
	
	/**
	 * Gets the time a connection will wait for a message without disconnecting. 
	 * @return time in milliseconds.
	 */
	int getTimeout();
	
	/**
	 * Sets the time a connection will wait for a message without disconnecting.
	 * @param value the time in milliseconds. Note: (0 is wait indefinite)
	 */
	void setTimeout(int value);

	/**
	 * Gets the remote endpoint this connection is connected to.
	 * @return the remote endpoint as a InetSocketAddress.
	 */
	InetSocketAddress getRemoteEndPoint();
	
	/**
	 * Gets the local endpoint this connection is connected on.
	 * @return the local endpoint as a InetSocketAddress.
	 */
	InetSocketAddress getLocalEndPoint();

	/**
	 * Closes the connection disposing any used resources.
	 */
	void close();
}